### ISSUE TITLE
Feature Request Variable Length 

### ISSUE THREAD

**Issue by pertschuk** [tqdm/tqdm#919]
- [ X] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] visual output bug
    + [ ] documentation request (i.e. "X is missing from the documentation." If instead I want to ask "how to use X?" I understand [StackOverflow#tqdm] is more appropriate)
    + [X] new feature request
- [ ] I have visited the [source website], and in particular
  read the [known issues]
- [X] I have searched through the [issue tracker] for duplicates
- [ ] I have mentioned version numbers, operating system and
  environment, where applicable:
  ```python
  import tqdm, sys
  print(tqdm.__version__, sys.version, sys.platform)
  ```

  [source website]: https://github.com/tqdm/tqdm/
  [known issues]: https://github.com/tqdm/tqdm/#faq-and-known-issues
  [issue tracker]: https://github.com/tqdm/tqdm/issues?q=
  [StackOverflow#tqdm]: https://stackoverflow.com/questions/tagged/tqdm

I'm working on making the tqdm progress bar in [Pytorch Lightning](https://github.com/PyTorchLightning/pytorch-lightning) accurate

The problem is it has both training (longer) and eval (shorter) steps in the same main progress bar. 

Would be really super helpful if I could pass a list of tuples to tqdm like [(num_train_samples, init_avg_time_per_train), (num_eval_sample, init_avg_time_per_eval) ...] and tqdm would keep track of the partitioned set, updating average times.

I could calculate this percentage manually, and then update tqdm on a say 100 point scale, but then tqdm would no longer show index/total_examples which is necessary. 

Thoughts? Thanks

**Comment by casperdcl** [comment]
Haven't checked but are you essentially talking about:

```python
from tqdm import tqdm
from time import sleep

with tqdm(desc="alternating deltas", total=40) as t:
  for _ in range(20):
    sleep(1)
    t.update()
    sleep(0.1)
    t.update()
```

**Comment by pertschuk** [comment]
@casperdcl Yeah. So right now if I do that and scale the update() calls it looks like 

`88.87939723629906/1897900`

 if 1897900 is total batch size.

But i would want it to say the correct batch, just scale the time accordingly

**Comment by casperdcl** [comment]
Hmm. I'd argue tracking two separate things in one bar is bad practice. If you really don't want to use nested bars and also don't want to track the two together (`total=20; sleep(1); sleep(0.1); update()`) then maybe just use `smoothing=0` which would pretty quickly give a reasonable ETA.
